src/Data/Subscription/SubscriptionDataClass.ts

```diff
@@ -0,0 +1,93 @@
+import {
+  ClientError,
+  getInstanceId,
+  provideDataClass,
+  unstable_JrRestApi,
+} from 'scrivito'
+
+interface Topic {
+  id: string
+  title: string
+  description?: string
+}
+
+export const Subscription = provideDataClass('Subscription', {
+  connection: {
+    async index() {
+      const myTopics = (
+        sanitizeResults(
+          await unstable_JrRestApi.fetch(
+            `neoletter/instances/${getInstanceId()}/my/topics`,
+          ),
+        ) as { results: Topic[] }
+      ).results
+
+      const mySubscribedTopicIds = await fetchMySubscribedTopicIds()
+
+      return {
+        results: myTopics.map(({ id, description, title }) => ({
+          id,
+          consent: mySubscribedTopicIds.includes(id) ? 'given' : 'notGiven',
```

Comment:
Since we are now showing/updating the value using the `DataFormBooleanWidget` we can switch back to a real boolean here:

```suggestion
          consent: mySubscribedTopicIds.includes(id)
```

This would also make the `dataForNo` and `dataForYes` option in this PR obsolete.

Please also adjust the `consent` flag in the `GeneralMarketingConsent` accordingly.